src/unity/python/turicreate/test/test_prototype_models.py

```diff
@@ -0,0 +1,31 @@
+import turicreate as tc
+import unittest
+import tempfile
+import shutil
+
+class SparseNNTest(unittest.TestCase):
+
+    def test_sparse_nn(self):
+        X = tc.util.generate_random_sframe(100, "ssszzz")
+        Y = X.copy()
+        X = X.add_row_number()
+
+        m = tc.extensions._sparse_nn()
+
+        m.train(X, "id")
```

Comment:
Shouldn't we try to mirror the standard model APIs here, with a static function `create` rather than an instance function `train`?

Reply:
Unlike in python, the C++ classes have to be instantiated in many situations where they are not trained (e.g. just to get the name) and need a basically constructor.  We could do this, but it should be something separate.  Let's discuss but keep this for now.

Reply:
Oh, right - I forgot about that. Let's keep that in mind for the long term, but for now this is fine.